packages/runner/src/result-emitter.ts

```diff
@@ -0,0 +1,125 @@
+import { ChildProcessWithoutNullStreams } from "child_process";
+import EventEmitter from "events";
+import { Readable } from "stream";
+
+import {
+  TestCaseType,
+  UserResult,
+  isEncodedResultError,
+} from "@alg-wiki/types";
+
+import { Boundary } from "./boundary";
+
+export enum ResultEmitterEvent {
+  Result = "result",
+  Stdout = "stdout",
+  Stderr = "stderr",
+  Close = "close",
+}
+
+// TODO: type-safety when adding listeners (get callbacks to have the right types/args/etc)
```

Comment:
I couldn't figure out a way to get nice typed callback signatures. I didn't try that hard either 😆 but if you have any ideas shout out.

Reply:
Another TODO is to limit the amount of output to a max size